### PR TITLE
f-metadata@2.4.0: Add in-app message callback

### PR DESCRIPTION
**Pull request by asherstoppard**
Allow clients to intercept in-app messages to push data to external applications (such as GA).

## UI Review Checks

- [x] README and/or UI Documentation has been [created|updated]
- [x] Unit tests have been [created|updated]
- N/A This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

### CODE REVIEW

**Review by ashleynolan — DISMISSED** [review]

File: packages/f-metadata/tests/index.test.js [hunk 1]
```diff
@@ -89,20 +97,25 @@ describe('f-metadata', () => {
...
+        expect.assertions(1);
         initialiseBraze(settings).then(() => {
             // Assert
             appboy.changeUser.mock.calls[0][1]();
             expect(push).toHaveBeenCalledWith({ event: 'appboyReady' });
         });
     });
 
-    it('should trigger the given callback when content cards have updated', () => {
+    it('should trigger the given callbacks when content cards have updated', () => {
```
Comment (ashleynolan) [inline packages/f-metadata/tests/index.test.js]: I'd probably split this test out now.

Such as `should call handleContentCards when cards are updated` and the new expects as a separate test (for intercepting app messages)

File: packages/f-metadata/tests/index.test.js [hunk 2]
```diff
@@ -89,20 +97,37 @@ describe('f-metadata', () => {
...
 
-    it('should trigger the given callback when content cards have updated', () => {
-        const cards = ['__CARD__'];
-        // Act
-        initialiseBraze(settings).then(() => {
-            // Assert
-            appboy.subscribeToContentCardsUpdates.mock.calls[0][0](cards);
-            expect(handleContentCards).toHaveBeenCalledWith(cards);
+    describe('callbacks', () => {
+        it('should call `handleContentCards` content cards have updated', () => {
```
Comment (kevinrodrigues) [inline packages/f-metadata/tests/index.test.js]: Not sure this one read right? Or is it just me :)

Comment (asherstoppard) [reply packages/f-metadata/tests/index.test.js]: Nope you're absolutely right 😆 I've pushed a change

File: packages/f-metadata/tests/index.test.js [hunk 3]
```diff
@@ -63,18 +67,22 @@ describe('f-metadata', () => {
...
     it('should initialise appboy and setup relevant settings', () => {
         // Assemble & Act
-        initialiseBraze(settings).then(() => {
-            // Assert
-            expect(appboy.initialize).toHaveBeenCalledWith(apiKey, { enableLogging, sessionTimeoutInSeconds });
-            expect(appboy.display.automaticallyShowNewInAppMessages).toHaveBeenCalled();
-            expect(appboy.openSession).toHaveBeenCalled();
-        });
+        expect.assertions(2);
+        initialiseBraze(settings)
```
Comment (xander-marjoram) [inline packages/f-metadata/tests/index.test.js]: Could we instead make these async tests and await the outcome of `initialiseBraze` before asserting?

File: packages/f-metadata/CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,14 @@
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+2.4.0
+------------------------------
+*May  8, 2020*
```
Comment (xander-marjoram) [inline packages/f-metadata/CHANGELOG.md]: Nit: If you do end up pushing another commit, can you update the date, fix this double space and remove the empty line between `### Changed`?

**Comment by ashleynolan** [comment]
@asherstoppard Looks like the build is failing because of babel issues running the tests (it's an `async > await` issue in jest).

If you add the [following setup to the babel config](https://github.com/justeat/fozzie-components/blob/master/packages/f-header/babel.config.js#L16), this error should go away.

**Comment by asherstoppard** [comment]
> @asherstoppard Looks like the build is failing because of babel issues running the tests (it's an `async > await` issue in jest).
> 
> If you add the [following setup to the babel config](https://github.com/justeat/fozzie-components/blob/master/packages/f-header/babel.config.js#L16), this error should go away.

@ashleynolan Apologies on the delay. Just pushing a change now, thanks for that 🙂

File: packages/f-metadata/babel.config.js [hunk 5]
```diff
@@ -8,12 +8,13 @@ module.exports = api => {
 
     if (isTest) {
         plugins.push('dynamic-import-node'); // to support dynamic imports (and webpack dynamic imports) when running unit tests
+        presets.push(['@babel/env', { targets: { node: 'current' } }]);
     } else {
         api.cache(true);
+        presets.push('@babel/env');
     }
 
     // use for both test and dev/live
```
Comment (ashleynolan) [inline packages/f-metadata/babel.config.js]: Can remove this comment now too

**Review by ashleynolan — APPROVED** [review]

**Review by kevinrodrigues — APPROVED** [review]
